torchao/quantization/qat/utils.py

```diff
@@ -104,3 +106,42 @@ def _get_qmin_qmax(n_bit: int, symmetric: bool = True):
         qmin = 0
         qmax = 2**n_bit - 1
     return (qmin, qmax)
+
+
+# log deprecation warning only once per class
+_LOGGED_DEPRECATED_CLASSES = set[type]()
+
+
+def _log_deprecation_warning(old_api_object: Any):
+    """
+    Log a helpful deprecation message pointing users to the new QAT API,
+    only once per deprecated class.
+    """
+    global _LOGGED_DEPRECATED_CLASSES
+    if old_api_object.__class__ in _LOGGED_DEPRECATED_CLASSES:
+        return
+    _LOGGED_DEPRECATED_CLASSES.add(old_api_object.__class__)
+    logger = logging.getLogger(old_api_object.__module__)
+    logger.warning(
+        """'%s' is deprecated and will be removed in a future release. Please use the following API instead:
```

Comment:
is there a concrete time for deprecation?

Reply:
yeah it's in the issue, deprecated this version (0.13.0) and removed the next (0.14.0)